.github/workflows/data-generator.yml

```diff
@@ -0,0 +1,68 @@
+# Defines a workflow that generates large volumes of real-ish data to work with for various development tasks.
+
+name: Data Generator
```

Comment:
Data Generator is a bit too generic IMO. This generates data for profiling, so maybe `Profiling Test Data Generator`?